src/worker.cpp

```diff
@@ -1169,24 +1169,44 @@ types::VPDMapVariant Worker::parseVpdFile(const std::string& i_vpdFilePath)
 
     std::shared_ptr<Parser> vpdParser = std::make_shared<Parser>(i_vpdFilePath,
                                                                  m_parsedJson);
-    types::VPDMapVariant l_parsedVpd = vpdParser->parse();
-
-    // Before returning, as collection is over, check if FRU qualifies for
-    // any post action in the flow of collection.
-    // Note: Don't change the order, post action needs to be processed only
-    // after collection for FRU is successfully done.
-    if (jsonUtility::isActionRequired(m_parsedJson, i_vpdFilePath, "postAction",
-                                      "collection"))
+    try
     {
-        if (!processPostAction(i_vpdFilePath, "collection", l_parsedVpd))
+        types::VPDMapVariant l_parsedVpd = vpdParser->parse();
+
+        // Before returning, as collection is over, check if FRU qualifies for
+        // any post action in the flow of collection.
+        // Note: Don't change the order, post action needs to be processed only
+        // after collection for FRU is successfully done.
+        if (jsonUtility::isActionRequired(m_parsedJson, i_vpdFilePath,
+                                          "postAction", "collection"))
         {
-            throw std::runtime_error("Required post action failed for path " +
-                                     i_vpdFilePath +
-                                     " Aborting collection for this FRU");
+            if (!processPostAction(i_vpdFilePath, "collection", l_parsedVpd))
+            {
+                throw std::runtime_error(
+                    "Required post action failed for path " + i_vpdFilePath +
+                    " Aborting collection for this FRU");
```

Comment:
Why we are aborting the collection if "postAction" fails ? 
why we are not populating the collected VPD ?

Reply:
I didn't change this logic. I just moved it under `try` block.

Reply:
@branupama Why do you think that a FRU should be populated if there is a processing failure at any stage?

Reply:
Going with the current requirements I guess, I would agree that we should populate collected data.
@souvik1914581 Could you please convert that throw at line 1185 to a //TODO PEL in this case and just log what we are throwing and continue with the flow instead of stopping it.

Reply:
Done